be/src/storage/lake/transactions.cpp

```diff
@@ -49,7 +93,25 @@ StatusOr<TabletMetadataPtr> publish_version(TabletManager* tablet_mgr, int64_t t
         return base_metadata_or.status();
     }
 
+    int base_version_index = -1;
+    auto index_status = get_base_tablet_metadata_index(tablet_mgr, tablet_id, base_version, new_version, txn_ids,
```

Comment:
This function may introduce additional object storage access. When batch size is 1, which should be common in low concurrency environments, it will cause negative optimization. The apply for loop below also has logic to check txn log existence, so is it possible to combine this function's logic with the apply for loop below, to avoid extra object storage access?

Reply:
I understand what's you mearning is that there will be a one extra object storage access in txn_log of `base_version_index`, but it likely hit cache when judge whether txn_log  exists the second time？Or i can add `if` in `for` logic